File: Trabalho Final/agenda.py
class Agenda:
  def __init__(self):
    self.datas_alugado = []
    self.datas_bloqueado = []

  # GETs
  def get_datas_alugado(self):
    return self.datas_alugado

  def get_datas_bloqueado(self):
    return self.datas_bloqueado

  # SETs
  def adiciona_data_alugado(self, data):
    self.datas_alugado.append(data)

  def remove_data_alugado(self, data):
    if data in self.datas_alugado:
      self.datas_alugado.remove(data)

  def adiciona_data_bloqueado(self, data):
    self.datas_bloqueado.append(data)

  def remove_data_bloqueado(self, data):
    if data in self.datas_bloqueado:
      self.datas_bloqueado.remove(data)

File: Exercício Incremental/05/Agenda.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Agenda
{
    // Se não tiver alugado nem bloqueado é porque está disponível
    private ArrayList<LocalDate> datasAlugado;
    private ArrayList<LocalDate> datasBloqueado;

    public Agenda()
    {
        this.datasAlugado = new ArrayList<LocalDate>();
        this.datasBloqueado = new ArrayList<LocalDate>();
    }

    // GETs    
    public ArrayList<LocalDate> getDatasAlugado() {
        return this.datasAlugado;
    }
    
    public ArrayList<LocalDate> getDatasBloqueado() {
        return this.datasBloqueado;
    }
    
    // SETs   
    public void adicionaDataAlugado (LocalDate data) {
        this.datasAlugado.add(data);
    }
    public void removeDataAlugado (LocalDate data) {
        if (this.datasAlugado.contains(data)) {
            this.datasAlugado.remove(data);
        }
    }
    
    public void adicionaDataBloqueado(LocalDate data) {
        this.datasBloqueado.add(data);
    }
    public void removeDataBloqueado(LocalDate data) {
        if (this.datasBloqueado.contains(data)) {
            this.datasBloqueado.remove(data);
        }            
    }
}
